activate.ps1

```diff
@@ -11,6 +11,11 @@ if ($MyInvocation.CommandOrigin -eq 'runspace') {
     exit 1
 }
 
+function dev-tye() {
+    $ProjectPath = Join-Path (Join-Path $PSScriptRoot "src") "tye"
+    dotnet run --project "$ProjectPath" @args
```

Comment:
There is a performance cost for doing dotnet run vs running the executable directly if it’s already built and doesn’t need update.

Reply:
Correct, but it's also the only way to ensure that if you made a change to tye, you're getting it. If we use the exe directly, it could be out-of-date relative to the code.

Reply:
I don't disagree, but it's going to feel significantly slower, especially if we want to use a dev version of tye for demoing. 

For example, try running tye --version with dotnet run vs running the exe. It's going to feel much better with the exe, even though it doesn't build

Maybe we need two options here. One that builds and one that doesn't build.

Reply:
It's "a *lot* slower" when it needs to rebuild, and just "slower" when there hasn't been a code change. To do something like this, you have to compare two time losses:

1. Time spent determining if a rebuild is needed (which is what `dotnet run` does)
2. The expected time spent by engineers using the script and not realizing it's using an out-of-date build.

In my extremely anecdotal past history, I actually find the latter more costly than the former. Even if it doesn't actually waste more time, it builds up my frustration level by making me say "**why isn't my code working!?!?!**" more often :).

So from my perspective, I'm OK losing a bit of time with each command. It's also a relatively constant time increase. So if you're doing a `dev-tye run`, for example, the percent of time lost to `dotnet run` is much smaller than when you're doing `dev-tye --version`, because the `tye run` command takes longer to run anyway and the build is a relatively-constant time operation.

> Maybe we need two options here. One that builds and one that doesn't build.

I'm fine with that, but I'd always use the one that builds 😄. Either way, it's y'alls repo and I'm quite happy to defer to you for preferred behavior (I can always have my own script 😉)

Reply:
@rynowak @davidfowl thoughts?

Reply:
I would prefer this do `dotnet run`. It's trivial to install a private build if you need *teh fast*.

Reply:
I can tell you that if there's a flavor of this script that doesn't do a build, I will never never never never never use it. All of the productivity of running 1 second faster is erased the first time you have to stop and wonder if it's actually using the right build 😆